src/main.py

```diff
@@ -1,125 +1,246 @@
 
-        # Remove bulletin posts
-        await reaction.message.delete()
+        # Skip the usual role check when trying to remove posts
+        # Users without a showcase role may still remove any of their posts from the showcase
+        linked_message = await get_linked_message(repost_message=message)
 
-@client.event
-async def on_message(message: discord.Message):
-    # Ignore all bots
-    if message.author.bot:
-        return
+        # Ignore remove attempts on broken links
+        if linked_message is None:
+            reply = strings.get("publish_error_remove").format(f"<@&{ADMIN_ROLE}>")
 
-    # Send posts from creative channel to verification channel
-    if message.channel.id in ART_CHANS:
-        await verify_art(message=message)
+        # Ignore interactions from users other than the message author
+        elif interaction.user != linked_message.author:
+            reply = strings.get("publish_error_remove_other")
+
+        # Remove showcase posts
+        else:
+            reply = strings.get("publish_response_remove_self")
+            success = True
+            await message.delete()
 
-async def verify_art(message: discord.Message):
-    verify = client.get_channel(VERIFY_CHAN)
+    # User interactions on posts in unhandled channels
+    else:
+        reply = strings.get("publish_error_channel")
+
+    # Send a secret reply to the commander
+    emoji = strings.emoji_success if success else strings.emoji_failure
+    if reply is None:
+        reply = strings.get("publish_error_generic")
+    await interaction.response.send_message(f"{emoji}\t{reply}", ephemeral=True)
+
+    # Add a reaction to the post to show it's been interacted with
+    if message is not None and (success or not fail_quietly):
+        await message.add_reaction(emoji)
+
+async def verify_art(message: discord.Message) -> None:
+    """
+    Creates a formatted message preview of a message for the bot to send in the verification channel.
+    :param message: The original message from a listening channel.
+    """
+    verify = bot.get_channel(VERIFY_CHAN)
     for img in message.attachments:
-        await verify.send(f"<@{message.author.id}> has posted:\n{message.content}\n{img.url}")
-
-async def publish_art(message: discord.Message):
-    gallery = client.get_channel(GALLERY_CHAN)
-    user = message.mentions[0]
-    title = f"Some amazing art by {user}"
-    split = message.content.split('\n')
-    text = split[-2]
-    embed = discord.Embed(title=title, type="rich", color=user.color, description=text)
+        await verify.send(strings.get("message_verify").format(f"<@{message.author.id}>", message.content, img.url, message.jump_url))
+
+async def publish_art(message: discord.Message) -> bool:
+    """
+    Creates a published message with embedded content for the bot to repost in the gallery channel.
+    """
+    gallery = bot.get_channel(GALLERY_CHAN)
+    author = await message.guild.fetch_member(message.raw_mentions[0])
```

Comment:
Why `raw_mentions` rather than `mentions`? `message.mentions[0]` should give us the user object and we can avoid calling the lookup function.

Reply:
Curiously when I was working through this I had an issue where `raw_mentions` was populated while `mentions` was not. Rather than wonder why that would ever be the case, I instead just went with `raw_mentions` instead. We could look into it again?

Reply:
It might have been caused by the bot having disabled mentions on init? If so, it's staying as raw mentions parsing.

Reply:
With mentions disabled, `clean_content` parses a mention as `@deleted-user`, `content` and `system_content` parse it as `'<@169338783307792385>`, `mentions` is `{list: 0} []`, and `raw_mentions` is `{list: 1} [169338783307792385]`.

With mentions enabled, `clean_content` parses a mention as `@rhubarb`, and `mentions` is `{list: 1} [<User id=169338783307792385 name='rhubarb' discriminator='4755' bot=False>]`.

So if we're fine with the bot pinging on messages using `user.mention` then we can use `mentions`, and otherwise we have to use `raw_mentions`.